application/leds.h

```diff
@@ -0,0 +1,29 @@
+/* Copyright Ville Hiltunen 2018 <hiltunenvillej@gmail.com>
+ *
+ * All code is public domain here.
+ */
+#ifndef LEDDRIVERH
+#define LEDDRIVERH
+
+#include "thread.h"
+#include "apa102.h"
+#include "apa102_params.h"
```

Comment:
I think it would be better to encapsulate the APA headers to the `.c`.   The user of the LEDs is not interested that they have been implemented with APA drivers.